src/app/utils/field-validation/validators/nricValidator.ts

```diff
@@ -0,0 +1,21 @@
+import { chain, left, right } from 'fp-ts/lib/Either'
+import { pipe } from 'fp-ts/lib/function'
+
+import { isNricValid } from 'src/shared/util/nric-validation'
+import { ResponseValidator } from 'src/types/field/utils/validation'
+import { ISingleAnswerResponse } from 'src/types/response'
+
+import { notEmptySingleAnswerResponse } from './common'
+
+type NricValidator = ResponseValidator<ISingleAnswerResponse>
+type NricValidatorConstructor = () => NricValidator
+
+const nricValidator: NricValidator = (response) => {
+  return isNricValid(response.answer)
+    ? right(response)
+    : left(`NricValidator:\tanswer is not a valid NRIC`)
```

Comment:
not sure if I'm missing something but did you mean 
``: left(`NricValidator:\t ${response.answer} is not a valid NRIC`)``

Reply:
both works la, one is more dynamic one is not. After the injection thingamagic I think having a generic error message is fine

Reply:
Actually, it is our policy to never log user-submitted answers for privacy reasons. FYI @karrui @mantariksh @tshuli